### PR TITLE
Praetorian Modsuit

### PR DESCRIPTION
**Pull request by Kyogon**
## About The Pull Request

Adds in a new, unique modsuit for Blueshields, the Praetorian Modsuit.

It is stored inside of the Blueshield's locker, so that it can be obtained on every map, even if they do not have an office or a suit storage machine.

The PR also adds it to the list of items removed during cryo, in the rare event that a blueshield decides to go to sleep wearing their new modsuit.

It is fully sprited, both inventory and worn, in a slick black and blue color.

## How This Contributes To The Skyrat Roleplay Experience

Blueshield, while a notable member of command staff as their protector, had the option of either a basic modsuit from robotics, or just a bag. I felt they deserved something befitting their role as a sort of honor guard for command.

This also allows them to, simply put, do their job of keeping command alive easier, while also standing apart from security due to the unique look. No one will mistake them for John Officer again!

## Proof of Testing

This generally was a pretty easy thing to make. I had some minor compilation issues, as well as some confusion in how to attach the sprites to the suit itself. 
At first, the suit was fully functional, but with no sprites. After some troubleshooting and figuring out where all of the pieces of code attached, I found out exactly where I needed to attach the sprites, largely using the contractor modsuit as an example. Armor values were also incorrect at first, but this was rectified, and confirmed on a test server.

I had no issues getting it to spawn in the lockers. That worked from the start.

I also edited the sprite halfway through, as originally the goal was to give them a larger blue pauldron, but in practice, it looked like crap. I am satisfied with where it currently stands, though once more people of different size and race start to wear it, I may update it once more.

Updated sprites shown below! 

image

image

## Changelog

:cl: Kyo
add: Adds a new, unique modsuit for Blueshields - The Praetorian Modsuit!
add: Adds the Praetorian modsuit to blueshield lockers.
code: Added the Praetorian modsuit to list of items removed from cryo, in case a sleepy blueshield forgets to take off their suit.
/:cl:

### CODE REVIEW

**Comment by desertmote** [comment]
This is dope, but the armor is too robust. My recommendation would be `mod_theme_safeguard` instead. Preferably everything about the magnate mod stays one-of-a-kind.

**Comment by Kyogon** [comment]
> This is dope, but the armor is too robust. My recommendation would be `mod_theme_safeguard` instead. Preferably everything about the magnate mod stays one-of-a-kind.

I can change this if necessary. When I was planning the suit with golden, though, this was what was discussed. Blueshield's armor is as good as this armor regardless. I'll talk to Golden and see what the consensus is.

**Comment by chemistrymain2** [comment]
The fireproof, acidproof Captain's Hardsuit which has the best available MODsuit armor should probably stay right where it is, yes. The blueshield's armor is as good, but it's not fireproof and acidproof and spaceproof.

**Comment by Kyogon** [comment]
> The fireproof, acidproof Captain's Hardsuit which has the best available MODsuit armor should probably stay right where it is, yes. The blueshield's armor is as good, but it's not fireproof and acidproof and spaceproof.

A good point, thinking of it. I will reduce the armor value to safeguard levels then based on the feedback you two gave.

**Comment by Namwty** [comment]
I may be wrong, I don't play blueshield, but last I checked, their jacked which only protects arms/chest didn't even have wounds armor and generally values weren't up to what modsuit provides. Not going to mention armor vest, which only blocks chest.

Modsuit protects all limbs, including legs and has ability to be suited with powerfull mods, like anomaly locks ones, would highly recommend reviewing the armor values. Not to mention protection against hazards.

Edit: didn't notice response, so ignore meh.

**Comment by Kyogon** [comment]
> I may be wrong, I don't play blueshield, but last I checked, their jacked which only protects arms/chest didn't even have wounds armor and generally values weren't up to what modsuit provides. Not going to mention armor vest, which only blocks chest.
> 
> Modsuit protects all limbs, including legs and has ability to be suited with powerfull mods, like anomaly locks ones, would highly recommend reviewing the armor values. Not to mention protection against hazards.
> 
> Edit: didn't notice response, so ignore meh.

I am keeping protection against hazards while reducing armor due to the nature of what Blueshield is. They usually will not be on the frontlines regardless, so keeping their strong point as the enhanced hazard protection above Safeguard is the goal. When planned out with Golden, it was agreed that this suit should be better than Safeguard but worse than Magnate for this reason. It simply allows them to get through hazards to recover heads of staff easier, it should rarely have an impact in actual combat.

**Comment by burgerenergy** [comment]
It's not the end of the world, but it proooobably shouldn't be named the same as a xenomorph thing for confusion's sake, given cm/tgmc's popularity in ss13

it is to late for me to come up with any alternatives rn tho

**Comment by Kyogon** [comment]
> It's not the end of the world, but it proooobably shouldn't be named the same as a xenomorph thing for confusion's sake, given cm/tgmc's popularity in ss13
> 
> it is to late for me to come up with any alternatives rn tho

I think it's fine for two reasons, one being that we will most likely never see Xenomorphs on this server so most people don't really expect to hear the word regardless, and two, praetorian is an ancient word for "bodyguard", specifically the royal bodyguard of the roman emperor.

Plus, no one really says the names of people's modsuits. No one calls the Captain "Magnate" or the HoS "Safeguard", so it won't cause issues.

File: code/modules/mod/mod_theme.dm
Comment (SpaceLoveSs13) [inline code/modules/mod/mod_theme.dm:1]: Why is this not modular?

**Review by lessthnthree — APPROVED** [review]

File: code/modules/mod/mod_theme.dm [hunk 1]
```diff
@@ -877,6 +877,74 @@
 	acid = 100
 	wound = 15
 
+/datum/mod_theme/praetorian
```
Comment (lessthnthree) [inline code/modules/mod/mod_theme.dm]: Please move this to modular_skyrat/modules/blueshield/code

File: code/modules/mod/mod_types.dm [hunk 2]
```diff
@@ -196,6 +196,21 @@
 		/obj/item/mod/module/jetpack/advanced,
 	)
 
+/obj/item/mod/control/pre_equipped/praetorian
```
Comment (lessthnthree) [inline code/modules/mod/mod_types.dm]: Please move this to modular_skyrat/modules/blueshield/code

**Review by lessthnthree — CHANGES_REQUESTED** [review]

**Comment by lessthnthree** [comment]
I think the name Praetorian is fine, but I'd prefer if the actual item path was /datum/mod_theme/blueshield, /obj/item/mod/control/pre_equipped/blueshield etc for ease of search in the code/spawn menu.

**Comment by Kyogon** [comment]
> I think the name Praetorian is fine, but I'd prefer if the actual item path was /datum/mod_theme/blueshield, /obj/item/mod/control/pre_equipped/blueshield etc for ease of search in the code/spawn menu.

Can do, I'm going to move on renaming and moving this (Not closing it, oops)

File: code/modules/mod/mod_theme.dm (hunk 1 above)
Comment (Kyogon) [reply code/modules/mod/mod_theme.dm]: Code made modular as requested

File: code/modules/mod/mod_types.dm (hunk 2 above)
Comment (Kyogon) [reply code/modules/mod/mod_types.dm]: Code made modular as requested

File: code/modules/mod/mod_theme.dm
Comment (Kyogon) [reply code/modules/mod/mod_theme.dm:1]: It is now!

**Review by lessthnthree — APPROVED** [review]

File: modular_skyrat/modules/blueshield/code/modsuit/mod_theme.dm [hunk 3]
```diff
@@ -0,0 +1,67 @@
+/datum/mod_theme/blueshield
+	name = "praetorian"
+	desc = "A prototype of the Magnate-class suit issued to station Blueshields, still boasting exceptional protection worthy of an honor guard."
+	extended_desc = "A prototype of the Magnate-class suit issued for use with the station blueshields, \
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/blueshield/code/modsuit/mod_theme.dm]:
```suggestion
	extended_desc = "A prototype of the Magnate-class suit issued for use with the station Blueshields, \
```

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]
The palettes of blue used in the headset, visor, chest and the rest of the MODsuit seem to all be different. Pick one (personally I'd take the one from the arm) and stick with it.

The visor itself can be different, but that color just doesn't really fit the "neon" color scheme that you'd have for a visor that's backlit. You'll want to go for a much lighter blue than that. Think of the syndicate green visor, but in blue.

**Comment by Kyogon** [comment]
> The palettes of blue used in the headset, visor, chest and the rest of the MODsuit seem to all be different. Pick one (personally I'd take the one from the arm) and stick with it.
> 
> The visor itself can be different, but that color just doesn't really fit the "neon" color scheme that you'd have for a visor that's backlit. You'll want to go for a much lighter blue than that. Think of the syndicate green visor, but in blue.

Sure, I had some ideas in mind to change the palette around, I can give it a try now. I'll probably use the arm blue, yea.

**Review by GoldenAlpharex — APPROVED** [review]
Seems good to me now, should be ready to go. Hopefully it won't cause too many issues with Blueshields constantly being outside, but we'll deal with that if it happens.

**Pull request merged**